test/test_adis16475_p.py

```diff
@@ -1,15 +1,121 @@
+import adi
 import pytest
 
 hardware = "adis16475"
 classname = "adi.adis16475"
 
 
+@pytest.mark.iio_hardware(hardware, True)
+@pytest.mark.parametrize("classname", [(classname)])
+@pytest.mark.parametrize("channel", [0])
+def test_adis16475_rx_data(test_dma_rx, iio_uri, classname, channel):
+    test_dma_rx(iio_uri, classname, channel, buffer_size=16)
+
+
+@pytest.mark.iio_hardware(hardware, True)
+def test_adis16475_filter_low_pass_3db_frequency(iio_uri):
+    adis16475 = adi.adis16475(uri=iio_uri)
+
+    for i in [10, 20, 40, 80, 164, 360, 720]:
+        adis16475.filter_low_pass_3db_frequency = i
+        assert adis16475.filter_low_pass_3db_frequency == i
+
+    for i in range(0, 10):
+        adis16475.filter_low_pass_3db_frequency = i
+        assert adis16475.filter_low_pass_3db_frequency <= 10
+
+    for i in range(11, 20):
+        adis16475.filter_low_pass_3db_frequency = i
+        assert adis16475.filter_low_pass_3db_frequency <= 20
+
+    for i in range(21, 40):
+        adis16475.filter_low_pass_3db_frequency = i
+        assert adis16475.filter_low_pass_3db_frequency <= 40
+
+    for i in range(41, 80):
+        adis16475.filter_low_pass_3db_frequency = i
+        assert adis16475.filter_low_pass_3db_frequency <= 80
+
+    for i in range(81, 164):
+        adis16475.filter_low_pass_3db_frequency = i
+        assert adis16475.filter_low_pass_3db_frequency <= 164
+
+    for i in range(165, 360):
+        adis16475.filter_low_pass_3db_frequency = i
+        assert adis16475.filter_low_pass_3db_frequency <= 360
+
+    for i in range(361, 720):
+        adis16475.filter_low_pass_3db_frequency = i
+        assert adis16475.filter_low_pass_3db_frequency <= 720
+
+
+@pytest.mark.iio_hardware(hardware, True)
+def test_adis16475_sample_rate(iio_uri):
+    adis16475 = adi.adis16475(uri=iio_uri)
+
+    for i in range(1, 2000):
+        adis16475.sample_rate = 2000.0 / (i + 1)
+        assert adis16475.sample_rate >= (2000.0 / (i + 1) - 1)
+        assert adis16475.sample_rate <= (2000.0 / (i + 1) + 1)
+    adis16475.sample_rate = 2000.0
```

Comment:
Is this needed for downstream tests? If so either use a fixture to create a test dependency